File: Harm-Up/Sales-by-match/sales_by_match.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Result
{
    /*
     * Complete the 'sockMerchant' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER_ARRAY ar
     */

    public static int sockMerchant(int n, List<int> ar)
    {
        var unmatched = new HashSet<int>();
        var pairs = 0;

        foreach (var color in ar)
        {
            if (!unmatched.Add(color))
            {
                unmatched.Remove(color);
                pairs++;
            }
        }

        return pairs;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> ar = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arTemp => Convert.ToInt32(arTemp)).ToList();

        int result = Result.sockMerchant(n, ar);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Harm-Up/Sales-by-match/sales_by_match.py
#!/bin/python3

import os
import sys

def _demo_test():
    input_data = """9
10 20 20 10 10 30 50 10 20
"""
    return input_data.strip().split('\n')

def sockMerchant(n, ar):
    unmatched = set()
    pairs = 0

    for color in ar:
        if color in unmatched:
            unmatched.remove(color)
            pairs += 1
        else:
            unmatched.add(color)

    return pairs


if __name__ == '__main__':
    # output_path = os.environ.get('OUTPUT_PATH')
    # fptr = open(output_path, 'w') if output_path else sys.stdout
    fptr = sys.stdout


    # n = int(input().strip())
    # ar = list(map(int, input().rstrip().split()))

    demo_lines = _demo_test()
    n = int(demo_lines[0].strip())
    ar = list(map(int, demo_lines[1].strip().split()))
    
    result = sockMerchant(n, ar)

    fptr.write(str(result) + '\n')
    print(f"Expected 3 got {result}")
    
    if fptr is not sys.stdout:
        fptr.close()
